lib/Onyx.js

```diff
@@ -405,7 +395,7 @@ function evictStorageAndRetry(error, ionMethod, ...args) {
  */
 function set(key, val) {
     // Write the thing to persistent storage, which will trigger a storage event for any other tabs open on this domain
-    return AsyncStorage.setItem(key, JSON.stringify(val))
+    return OnyxInternal.set(key, val)
         .then(() => keyChanged(key, val))
         .catch(error => evictStorageAndRetry(error, set, key, val));
 }
```

Comment:
Initially I wanted to move this whole function to `Onyx.internals` as well, but when I did I had to also move `keyChanged` and `evictStorageAndRetry` as well, as I moved those 2, I needed to also move, more and more functions and static variables, so I decided to just move the async part.